Python/pylifecycle.c

```diff
@@ -1450,8 +1467,10 @@ new_interpreter(PyThreadState **tstate_p)
     }
     _PyRuntimeState *runtime = &_PyRuntime;
 
-    if (!runtime->initialized) {
```

Comment:
I'd not remove this condition `initialized` is separate from `allow_new`, so having this check here still makes sense.

Reply:
Sorry for any confusion, @soltysh.  `_PyRuntimeState,interpreters.allow_new` was added in this PR to solve the problem of interpreters being created when they shouldn't be (e.g. during runtime finalization).  You could say the flag specifically means "new_interpreter() can be called currently". :smile:  So the change here is correct.

Reply:
@ericsnowcurrently so runtime is always a single object (like an uber-object) and then you can create multiple interpreters. Right, but that still requires the runtime to be initialized. Even though the situation should not happen, because I'd assume the first invocation of python would initialize the runtime, it should not hurt having this here. Unless my thinking is wrong here.

Reply:
I suppose your suggestion @soltysh would be to have two separate checks on `runtime->initialized` and `runtime->interpreters.allow_new`?

It's a while ago now, but I think my reasoning here was that this 'allow_new' flag encapsulates all information about whether a new interpreter can be created, so there should be no need to check things like whether the runtime is initialised (since 'allow_new' is only set to true when the runtime is initialised). Does that sounds reasonable?

I could be persuaded to change this if there are better suggestions :)